src/components/HotspotConfigurationPicker.tsx

```diff
@@ -0,0 +1,228 @@
+  onAntennaUpdated,
+  onGainUpdated,
+  onElevationUpdated,
+  outline,
+}: Props) => {
+  const { t } = useTranslation()
+  const colors = useColors()
+
+  const gainInputRef = useRef<TextInput | null>(null)
+  const elevationInputRef = useRef<TextInput | null>(null)
+
+  const [elevation, setElevation] = useState<string>()
+  const [gain, setGain] = useState<string>(
+    selectedAntenna.gain.toLocaleString(locale, {
+      maximumFractionDigits: 1,
+      minimumFractionDigits: 1,
+    }),
+  )
+
+  const onSelectAntenna = (value: string | number, index: number) => {
+    const antenna = Object.values(Antennas)[index]
+    onAntennaUpdated(antenna)
+    onGainUpdated(antenna.gain)
+    setGain(
+      antenna.gain.toLocaleString(locale, {
+        maximumFractionDigits: 1,
+        minimumFractionDigits: 1,
+      }),
+    )
+  }
+
+  const showElevationInfo = () =>
+    Alert.alert(
+      t('antennas.elevation_info.title'),
+      t('antennas.elevation_info.desc'),
+    )
+  const showGainInfo = () =>
+    Alert.alert(t('antennas.gain_info.title'), t('antennas.gain_info.desc'))
+
+  const focusGain = () => {
+    gainInputRef.current?.focus()
+  }
+  const focusElevation = () => {
+    elevationInputRef.current?.focus()
+  }
+
+  const onChangeGain = (text: string) => setGain(text)
+  const onDoneEditingGain = () => {
+    const gainFloat = parseFloat(
+      gain.replace(groupSeparator, '').replace(decimalSeparator, '.'),
+    )
+    let gainString
+    if (!gainFloat || gainFloat <= 1) {
+      gainString = '1'
+    } else if (gainFloat >= 15) {
+      gainString = '15'
+    } else {
+      gainString = gainFloat.toLocaleString(locale, {
+        maximumFractionDigits: 1,
+      })
+    }
+    setGain(gainString)
+    onGainUpdated(
+      parseFloat(
+        gain.replace(groupSeparator, '').replace(decimalSeparator, '.'),
+      ),
+    )
+  }
+
+  const onChangeElevation = (text: string) => setElevation(text)
+  const onDoneEditingElevation = () => {
+    const elevationInteger = elevation
+      ? parseInt(
+          elevation.replace(groupSeparator, '').replace(decimalSeparator, '.'),
+          10,
+        )
+      : 0
+    let stringElevation
+    if (!elevationInteger) {
+      stringElevation = '0'
+    } else {
+      stringElevation = elevationInteger.toString()
+    }
+    setElevation(stringElevation)
+    onElevationUpdated(parseInt(stringElevation, 10))
+  }
+
+  useEffect(() => {
+    setGain(
+      selectedAntenna.gain.toLocaleString(locale, {
+        maximumFractionDigits: 1,
+        minimumFractionDigits: 1,
+      }),
+    )
+  }, [selectedAntenna])
+
+  return (
+    <Box
+      backgroundColor="white"
+      borderRadius="m"
+      marginVertical="l"
+      borderWidth={outline ? 1 : 0}
+      borderColor="grayLight"
+    >
+      <HeliumActionSheet
+        title={t('antennas.onboarding.select')}
+        prefixVariant="regular"
+        prefixFontSize={14}
+        data={Object.values(Antennas).map((a) => ({
```

Comment:
Wouldn't hurt to move this data out to a `useMemo`

Reply:
Done